qiskit/pulse/instructions/delay.py

```diff
@@ -48,8 +49,13 @@ def __init__(
             duration: Length of time of the delay in terms of dt.
             channel: The channel that will have the delay.
             name: Name of the delay for display purposes.
+
+        Raises:
+            PulseError: If `channel` cannot be delayed because it cannot be scheduled.
         """
         super().__init__(operands=(duration, channel), name=name)
+        if not channel.is_schedulable:
```

Comment:
This can be written `if isinstance(channel, ClassicalIOChannel)` by introducing dedicated class. This would make it clearer what is prohibited. It would be great if you could insert this check to other instructions too. 

Perhaps, as @taalexander commented, we might need to consider delays on classical IO channels in relative scheduling models, i.e. `ScheduleBlock`. So we can drop channel check from `Delay` instruction, and add this check to other instructions such as `Play`, `ShiftPhase`, `SetPhase`, `ShiftFrequency`, and `SetFrequency`, which will never take classical IO channels.

Reply:
> It would be great if you could insert this check to other instructions too.

I'm not totally sure which other instructions it should apply to. I've gone through the ones in this namespace and have thrown this error for instructions which have channels that aren't already restricted to a subtype of Channel but that has broken some tests e.g. RelativeBarrier has a test which takes a MemorySlot as one of the channels in its constructor.